python/main.py

```diff
@@ -25,10 +28,64 @@ def root():
 
 
 @app.post("/items")
-def add_item(name: str = Form(...)):
+def add_item(name: str = Form(...),category: str =Form(...),image_path: str= Form(...)):
     logger.info(f"Receive item: {name}")
+     # step1:check if the file exists and if not create an empty json list
+    if not items_file.exists() or not items_file.is_file():
+        items_data = {"items": []}
+    else:
+        # if file exists load the existing data
+        try:
+            with open(items_file, 'r') as file:
+                items_data = json.load(file)
+        except json.decoder.JSONDecodeError:
+            # handling case if the file content is not valid json
+            items_data = {"items": []}
+
+    #converting the image to image_hash
+    image_filename = os.path.basename(image_path)
+    image_hash = hashlib.sha256(image_filename.encode()).hexdigest()
+    # appending new item recieved
+    new_item = {"name": name, "category": category,"image":f"{image_hash}.jpg"}
+    items_data["items"].append(new_item)
+
+    # rewriting the file with updated data
+    with open(items_file, 'w') as file:
+        json.dump(items_data, file, indent=2)
+
     return {"message": f"item received: {name}"}
 
+@app.get("/items")
+def get_items():
+    if not items_file.exists() or not items_file.is_file():
+        items_data = {"items": []}
+    else:
+        # if file exists load the existing data
+        try:
+            with open(items_file, 'r') as file:
+                items_data = json.load(file)
+        except json.decoder.JSONDecodeError:
+            # handling case if the file content is not valid json
+            items_data = {"items": []}
+    return {"items": items_data}        
+
+@app.get("/items/{item_id}")
+def get_one_item(item_id : int = Path(..., title="The ID of the item to retrieve")):
+    if not items_file.exists() or not items_file.is_file():
+        items_data = {"items": []}
+    else:
+        # if file exists load the existing data
+        try:
+            with open(items_file, 'r') as file:
+                items_data = json.load(file)
+        except json.decoder.JSONDecodeError:
+            # handling case if the file content is not valid json
+            items_data = {"items": []}
+    if 0 <= item_id - 1 < len(items_data["items"]):
```

Comment:
I don't think this works like this 👀 . You should do more like item_id >= 0 and item_id < len(items_data["items"])